### PR TITLE
[fix](group commit) Fix group commit load complex data type in nereids planner

### PR DESCRIPTION
**Pull request by mymeiyi**
## Proposed changes

1. when use group commit in nereids planner to insert struct data type, may get `too many filtered rows`, because the converted string value is not correct.
2. Fix case: group commit can be executed in observer fe, the case try to get profile from observer to ensure the group commit is not forwarded, but no profiler can get, so the case fail.

### CODE REVIEW

**Comment by mymeiyi** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertUtils.java [hunk 1]
```diff
@@ -161,10 +161,16 @@ public static InternalService.PDataRow getRowStringValue(List<NamedExpression> c
...
-                row.addColBuilder().setValue(String.format("\"%s\"",
-                        ((Literal) expr).toLegacyLiteral().getStringValue()));
+                String stringValue = ((Literal) expr).toLegacyLiteral().getStringValueForStreamLoad();
+                if (stringValue.equals(StmtExecutor.NULL_VALUE_FOR_LOAD) || stringValue.startsWith("\"")
+                        || stringValue.endsWith(
+                        "\"")) {
+                    row.addColBuilder().setValue(String.format("\"%s\"", stringValue));
+                } else {
+                    row.addColBuilder().setValue(String.format("%s", stringValue));
+                }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertUtils.java:173]: add comment to explain, cannot understand what u want to do

**Comment by mymeiyi** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertUtils.java [hunk 2]
```diff
@@ -161,10 +164,16 @@ public static InternalService.PDataRow getRowStringValue(List<NamedExpression> c
...
             } else {
-                row.addColBuilder().setValue(String.format("\"%s\"",
-                        ((Literal) expr).toLegacyLiteral().getStringValue()));
+                String stringValue = ((Literal) expr).toLegacyLiteral().getStringValueForStreamLoad();
+                if (stringValue.equals(StmtExecutor.NULL_VALUE_FOR_LOAD) || stringValue.startsWith("\"")
+                        || stringValue.endsWith(
+                        "\"")) {
+                    row.addColBuilder().setValue(String.format("\"%s\"", stringValue));
+                } else {
+                    row.addColBuilder().setValue(String.format("%s", stringValue));
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertUtils.java:175]: could process string literal `abc"dbf` correctly?

**Pull request closed without merging**